pkg/services/store/entity/health.go

```diff
@@ -0,0 +1,52 @@
+package entity
+
+import (
+	"context"
+
+	grpcAuth "github.com/grpc-ecosystem/go-grpc-middleware/auth"
+	"google.golang.org/grpc/health/grpc_health_v1"
+)
+
+// Compile-time assertion
+var _ HealthService = &healthServer{}
+
+type HealthService interface {
+	grpc_health_v1.HealthServer
+	grpcAuth.ServiceAuthFuncOverride
+}
+
+type InitializableEntityServer interface {
+	EntityStoreServer
+	Init() error
+}
+
+func ProvideHealthService(server InitializableEntityServer) (grpc_health_v1.HealthServer, error) {
+	h := &healthServer{entityServer: server}
+	return h, nil
+}
+
+type healthServer struct {
+	entityServer InitializableEntityServer
+}
+
+// AuthFuncOverride for no auth for health service.
+func (s *healthServer) AuthFuncOverride(ctx context.Context, _ string) (context.Context, error) {
```

Comment:
This never gets called, so I've worked around it for now above in `authenticator.go` above. Still trying to figure out why its not getting called.